arviz/utils.py

```diff
@@ -50,14 +54,34 @@ def _var_names(var_names, data):
                 )
             )
 
+        excluded_vars = [
+            var[1:] for var in var_names if var.startswith("~") and var not in all_vars
+        ]
         if excluded_vars:
-            var_names = [i for i in all_vars if i not in excluded_vars]
-
-        existent_vars = np.isin(var_names, all_vars)
-        if not np.all(existent_vars):
+            if (filter == "like") or (filter == "regex"):
+                for pattern in excluded_vars:
+                    if pattern not in all_vars:
```

Comment:
I _think_ this should not be here (see below too) because it should be valid to use the name itself even if we are using filter, the important check is the check above of `var not in all_vars` when creating `excluded_vars`